smsapi/ProxyHTTP.cs

```diff
@@ -70,155 +64,75 @@ public Stream Execute(string uri, NameValueCollection data, RequestMethod method
+                client.UserAgent = authentication.GetClientAgentHeader();
+                client.Authenticator = GetAuthenticator();
+            }
 
-                var httpRequest = webRequest as HttpWebRequest;
-                if (httpRequest != null)
-                {
-                    httpRequest.UserAgent = authentication.GetClientAgentHeader();
-                }
+            var request = new RestRequest(ToMethod(method));
+            foreach (string key in data.Keys)
+            {
+                request.AddParameter(key, data[key]);
             }
 
-            if (RequestMethod.POST.Equals(method) || RequestMethod.PUT.Equals(method))
+            foreach (KeyValuePair<string, Stream> file in files)
             {
-                Stream stream;
-
-                if (files != null && files.Count > 0)
-                {
-                    webRequest.ContentType = "multipart/form-data; boundary=" + boundary;
-                    stream = PrepareMultipartContent(boundary, data, files);
-                }
-                else
-                {
-                    webRequest.ContentType = "application/x-www-form-urlencoded";
-                    stream = PrepareContent(data);
-                }
-
-                webRequest.ContentLength = stream.Length;
-
-                try
-                {
-                    stream.Position = 0;
-                    CopyStream(stream, webRequest.GetRequestStream());
-                    stream.Close();
-                }
-                catch (WebException e)
-                {
-                    throw new ProxyException(e.Message, e);
-                }
+                request.AddFile(file.Key, s => file.Value.CopyTo(s), file.Key, file.Value.Length);
             }
 
-            var response = new MemoryStream();
+            var responseStream = new MemoryStream();
+            request.ResponseWriter = s => s.CopyTo(responseStream);
 
             try
             {
-                CopyStream(webRequest.GetResponse().GetResponseStream(), response);
+                client.Execute(request);
             }
-            catch (WebException e)
+            catch (System.Exception e)
             {
-                throw new ProxyException("Failed to get response from " + webRequest.RequestUri, e);
+                throw new ProxyException("Failed to get response from " + client.BuildUri(request), e);
             }
 
-            response.Position = 0;
-            return response;
+            return responseStream;
         }
 
-        protected Stream PrepareContent(NameValueCollection data)
+        private static Method ToMethod(RequestMethod method)
         {
-            Stream stream = new MemoryStream();
-
-            IEnumerator enumerator = data.GetEnumerator();
+            switch (method)
+            {
+                case RequestMethod.DELETE:
+                    return Method.DELETE;

```

Comment:
This can be assigned directly in enum.

Reply:
https://github.com/smsapi/smsapi-csharp-client/pull/31#discussion_r1068770536